build/lib/i18n.ts

```diff
@@ -308,4 +542,556 @@ export function processNlsFiles(opts:{fileHeader:string;}): ThroughStream {
+
+	if (ModuleJsonFormat.is(json)) {
+		extension.xlf.addFile(originalFilePath, json['keys'], json['messages']);
+	} else {
+		extension.xlf.addFile(originalFilePath, Object.keys(json), messages);
+	}
+
+	// Check if XLF is populated with file nodes to emit it
+	if (++extensions[extensionName].processed === localizationFilesCount) {
+		const newFilePath = path.join(projectName, extensionName + '.xlf');
+		const xlfFile = new File({ path: newFilePath, contents: new Buffer(extension.xlf.toString(), 'utf-8')});
+		stream.emit('data', xlfFile);
+	}
+}
+
+var islXlf: XLF,
+	islProcessed: number = 0;
+
+function importIsl(file: File, stream: ThroughStream) {
+	const islFiles = ['Default.isl', 'messages.en.isl'];
+	const projectName = 'vscode-workbench';
+
+	let xlf = islXlf ? islXlf : islXlf = new XLF(projectName),
+		keys: string[] = [],
+		messages: string[] = [];
+
+	let model = new TextModel(file.contents.toString());
+	let inMessageSection = false;
+	model.lines.forEach(line => {
+		if (line.length === 0) {
+			return;
+		}
+		let firstChar = line.charAt(0);
+		switch (firstChar) {
+			case ';':
+				// Comment line;
+				return;
+			case '[':
+				inMessageSection = '[Messages]' === line || '[CustomMessages]' === line;
+				return;
+		}
+		if (!inMessageSection) {
+			return;
+		}
+		let sections: string[] = line.split('=');
+		if (sections.length !== 2) {
+			log('Error:', `Badly formatted message found: ${line}`);
+		} else {
+			let key = sections[0];
+			let value = sections[1];
+			if (key.length > 0 && value.length > 0) {
+				keys.push(key);
+				messages.push(value);
+			}
+		}
+	});
+
+	const originalPath = file.path.substring(file.cwd.length+1, file.path.split('.')[0].length).replace(/\\/g, '/');
+	xlf.addFile(originalPath, keys, messages);
+
+	// Emit only upon all ISL files combined into single XLF instance
+	if (++islProcessed === islFiles.length) {
+		const newFilePath = path.join(projectName, 'setup.xlf');
+		const xlfFile = new File({ path: newFilePath, contents: new Buffer(xlf.toString(), 'utf-8')});
+		stream.emit('data', xlfFile);
+	}
+}
+
+export function pushXlfFiles(apiHostname: string, username: string, password: string): ThroughStream {
+	let tryGetPromises = [];
+	let updateCreatePromises = [];
+
+	return through(function(file: File) {
+		const project = path.dirname(file.relative);
+		const fileName = path.basename(file.path);
+		const slug = fileName.substr(0, fileName.length - '.xlf'.length);
+		const credentials = `${username}:${password}`;
+
+		// Check if resource already exists, if not, then create it.
+		let promise = tryGetResource(project, slug, apiHostname, credentials);
+		tryGetPromises.push(promise);
+		promise.then(exists => {
+			if (exists) {
+				promise = updateResource(project, slug, file, apiHostname, credentials);
+			} else {
+				promise = createResource(project, slug, file, apiHostname, credentials);
+			}
+			updateCreatePromises.push(promise);
+		}).catch((reason) => {
```

Comment:
I've made the pipeline to fail whenever we have any parsing or Transifex push/pull errors.